Headers/DebugServer2/Target/ThreadBase.h

```diff
@@ -61,6 +61,14 @@ class ThreadBase {
   virtual ErrorCode modifyRegisters(
       std::function<void(Architecture::CPUState &state)> action) final;
 
+#if defined(ARCH_X86) || defined(ARCH_X86_64)
```

Comment:
Here, adding `#if defined(OS_LINUX)` and make these pure-virtual would be good. That way we can enable them over time for each platform that gets an implementation, and if something is messed up, we get a link error instead of having to run the code to get a failure.